File: DotNetGotchas/CSharp/RTTI/RTTIExample/Trainer.cs
//Trainer.cs
using System;

namespace RTTIExample
{
	public class Trainer
	{
		public void Train(Animal anAnimal)
		{
			anAnimal.Eat();
			//Using casting

			Dog doggie = (Dog) anAnimal;
			doggie.Bark();

			Cat aCat = (Cat) anAnimal;
			aCat.Meow();
		}
	}
}

File: DotNetGotchas/VBDN/RTTI/RTTIExample/Trainer.vb
'Trainer.vb
Public Class Trainer
    Public Sub Train(ByVal anAnimal As Animal)
        anAnimal.Eat()

        'Using Casting

        Dim doggie As Dog = CType(anAnimal, Dog)
        doggie.Bark()

        Dim aCat As Cat = CType(anAnimal, Cat)
        aCat.Meow()
    End Sub
End Class
